rock_paper_scissors/api/models.py

```diff
@@ -0,0 +1,65 @@
+"""
+Relationships with Foreign Keys, esp. relationship() and FK values
+Why do we need to establish columns for players in game?
+difference with many to one and one to many -- if you back_populate either relationship type,
+    does it matter which you use?
+
+"""
+
+from sqlalchemy import Column, Integer, String
+from sqlalchemy import Enum
+from sqlalchemy import ForeignKey
+from sqlalchemy.ext.declarative import as_declarative
+from sqlalchemy.orm import relationship
+
+
+@as_declarative()
+class ModelBase(object):
+    def __tablename__(self):
+        return self.__class__.__name__.lower()
+
+
+class Game(ModelBase):
+    id = Column(Integer, primary_key=True)
+    player_1_id = Column(Integer, ForeignKey('user.id'))
+    player_2_id = Column(Integer, ForeignKey('user.id'))
+    round_id = Column(Integer, ForeignKey('round.id'))
+    rounds_choice = Column(Integer)
+    rounds_decrement = Column(Integer)
+    current_round = Column(Integer)
+    winner = Column(String, ForeignKey('user.id'))
+
+    round = relationship('Round', back_populates='game')
+    user = relationship('User', back_populates='game', uselist=False)
```

Comment:
Yes, we generally implement these for all connections between models in our systems. Part of the reason for this has to do with how we've orchestrated our APIs' representations of our our model-based objects.  Even aside from the API, though, it's usually very useful to have direct access to a related object because it's cuts out the step of needing to query for it; the relationship allows us to skip one or more lines of code, essentially.

I think you're fine regarding the SQLA nitty gritty, but be aware that this knowledge is _not_ universal. Other frameworks definitely have similarities to SQLAlchemy, but the "universal" bit here has to do with the database-level concepts of relational data modelling, not the ORM-specific functionality.